src/category_theory/natural_isomorphism.lean

```diff
@@ -110,7 +133,7 @@ lemma naturality_2 (α : F ≅ G) (f : X ⟶ Y) :
   (α.hom.app X) ≫ (G.map f) ≫ (α.inv.app Y) = F.map f :=
 begin erw [naturality, ←category.assoc, is_iso.hom_inv_id, category.id_comp] end
 
-def is_iso_of_is_iso_app (α : F ⟶ G) [∀ X : C, is_iso (α.app X)] : is_iso α :=
+instance is_iso_of_is_iso_app (α : F ⟶ G) [∀ X : C, is_iso (α.app X)] : is_iso α :=
```

Comment:
What's the reason for the choice of the instance markings here? My instinct would have been that `is_iso_app_of_is_iso` would be an instance and `is_iso_of_is_iso_app` would be a def though I don't have strong arguments either way - is there something I'm missing?

Reply:
I agree it is not obvious here. I've done it this way because it's convenient in the construction of `nat_iso.of_components` below. I could obviously rejigger than definition to not rely on any instances, and set this part up differently.

I'll try that now.

Reply:
Seems fine.